src/types/web-handler.ts

```diff
@@ -27,11 +31,174 @@ export type WithEdgeSpecRequestOptions<T> = T & EdgeSpecRequestOptions
 
 export type EdgeSpecRequest<T = {}> = WithEdgeSpecRequestOptions<Request> & T
 
-export type EdgeSpecResponse = Response
+export interface SerializableToResponse {
+  /**
+   *  Serialize the response to a Response object
+   *
+   * @throws z.ZodError if the response does not match the schema
+   * @param schema - the schema to validate the response against
+   */
+  serializeToResponse(schema: z.ZodTypeAny): Response
+}
+
+export type ValidFormDataValue = Primitive | Blob
+
+export abstract class EdgeSpecResponse implements SerializableToResponse {
```

Comment:
for these methods like `.json()`, what do you think about mirroring `Response` as much as possible instead of using builder functions?

e.g. instead of

```ts
EdgeSpecResponse.json({
  hello: "world",
}).status(201)
```

we could do

```ts
EdgeSpecResponse.json({
  hello: "world",
}, {
  status: 201
})
```

Reply:
both are currently possible

Reply:
ah my bad 👍